src/worker_http.rs

```diff
@@ -295,23 +304,16 @@ async fn async_pull_samples(
 
         // Append a new task to the queue
         tasks.push_back(tokio::spawn(pull_sample(
-            client.clone(),
+            shareable_client.clone(),
```

Comment:
the perf bug was here, we need to clone so that this moves into a task, but clone the raw client means that we share no connection. Boxing this prior to cloning is enough to restore the lost perf